site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+
+```ts
+app.post('/bindings', (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Use the user id and channel id to dynamically return certain UI elements. Though for this example, we won't use these value.
+    const channelID = call.context.channel_id;
+    const userID = call.context.acting_user_id;
+
+    const bindings: AppBinding[] = [{
+        location: '/post_menu',
+        bindings: [
+            {
+                location: 'create-ticket',
+                icon: 'create-ticket.png',
+                label: 'Create ticket',
+                call: {
+                    path: '/forms/create-ticket-post-menu',
+                    expand: {
+                        post: 'summary',
+                    },
+                },
+            },
+        ],
+    }];
+
+    const callResponse: AppCallResponse = {
+        type: 'ok',
+        data: bindings,
+    };
+
+    res.json(callResponse);
+});
+
+// When static files are fetched from our App, the request paths are prefixed with `/static`, so our files `icon.png` and `create-ticket.png` will be requested as `static/icon.png` and `static/create-ticket.png`. If we have all of our static assets in one directory, we can serve all of them like this:
+const staticDirectory = './static';
+app.use('/static', express.static(staticDirectory));
+```
+
+We now have an item in the post dropdown menu that says "Create ticket", along with the icon located at `static/create-ticket.png` displayed next to the label. When the post menu item is clicked, our App will receive a [Call]({{< ref "call" >}}) at the endpoint `/forms/create-ticket-from-post/submit`. If we return a form in our response, a modal will be shown to the user with our form in the modal.
+
+Note that the `expand` object with `post: 'summary'` means that we will receive the relevant `Post` at `call.context.post` when we receive the request. This makes it so we don't have to fetch the `Post` manually when handling the request. Learn more about Expand [here]({{< ref "call" >}}).
+
+By defining the post menu item's `Call` with the path `/forms/create-ticket-post-menu`, we've implicitly registered an endpoint to process a request when this item is clicked:
+- `/forms/create-ticket-post-menu/submit`
+
+### Opening a modal
+
+Now let's process the post menu item click, and return an `AppForm` to show as a modal:
+
+```ts
+app.post('/forms/create-ticket-post-menu/submit', (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Extract the Post from the expanded context
+    const post: Post = call.context.post;
+
+    const call: AppCall = {
+        path: '/forms/create-ticket',
+        expand: {
+            acting_user: 'summary', // Expand the acting user so we can get the user's username for our post
```

Comment:
As I said before, I feel these clarifications should not be made in code, but outside the code.